File: src/main/java/com/randomperson22/piggymodtest/npc/zizzy/EntityZizzyNpc.java
package com.randomperson22.piggymodtest.npc.zizzy;

import com.randomperson22.piggymodtest.npc.EntityNpc;
import net.minecraft.init.Blocks;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;

public class EntityZizzyNpc extends EntityNpc {
    public EntityZizzyNpc(World worldIn) {
        super(worldIn);
    }

    @Override
    protected Item getRequiredItem() {
        return null;
    }

    @Override
    protected boolean isCorrectItem(ItemStack stack) {
        return !stack.isEmpty()
            && stack.getItem() == Item.getItemFromBlock(Blocks.TALLGRASS)
            && stack.getMetadata() == 1;
    }

    @Override
    public void onUpdate() {
        super.onUpdate();
    }
}

File: src/main/kotlin/com/randomperson22/piggymodtest/npc/zizzy/EntityZizzyNpc.kt
package com.randomperson22.piggymodtest.npc.zizzy

import com.randomperson22.piggymodtest.npc.EntityNpc
import net.minecraft.init.Blocks
import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.world.World

class EntityZizzyNpc(worldIn: World) : EntityNpc(worldIn) {
    override fun getRequiredItem(): Item? {
        return null
    }

    override fun isCorrectItem(stack: ItemStack): Boolean {
        return !stack.isEmpty() && stack.getItem() === Item.getItemFromBlock(Blocks.TALLGRASS) && stack.getMetadata() == 1
    }

    override fun onUpdate() {
        super.onUpdate()
    }
}